File: ios/DailyMemory/Domain/UseCase/Person/UpdatePersonUseCase.swift
import Foundation

/// Use case for updating an existing person.
final class UpdatePersonUseCase {
    private let personRepository: PersonRepository

    init(personRepository: PersonRepository = PersonRepositoryImpl()) {
        self.personRepository = personRepository
    }

    func execute(_ person: Person) async throws -> Person {
        try await personRepository.update(person)
        return person
    }
}

File: android/app/src/main/java/com/dailymemory/domain/usecase/person/UpdatePersonUseCase.kt
package com.dailymemory.domain.usecase.person

import com.dailymemory.domain.model.Person
import com.dailymemory.domain.repository.PersonRepository
import javax.inject.Inject

/**
 * Use case for updating an existing person.
 */
class UpdatePersonUseCase @Inject constructor(
    private val personRepository: PersonRepository
) {
    suspend operator fun invoke(person: Person): Result<Person> {
        return try {
            personRepository.update(person)
            Result.success(person)
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}
